docs/aws/audit/dynamodbmonitoring/rules/sagemaker_notebook_direct_internet_access_remediation.mdx

```diff
@@ -1,6 +1,275 @@
+
+2. List all DynamoDB tables: Use the following AWS CLI command to list all the DynamoDB tables in your account:
+
+   ```
+   aws dynamodb list-tables
+   ```
+   This command will return a list of all DynamoDB tables in your account.
+
+3. Describe the DynamoDB table: For each table, you can use the following AWS CLI command to get detailed information about the table:
+
+   ```
+   aws dynamodb describe-table --table-name <YourTableName>
+   ```
+   Replace `<YourTableName>` with the name of your DynamoDB table. This command will return a JSON object that contains detailed information about the table.
+
+4. Check for Notebook Direct Internet Access: Unfortunately, DynamoDB does not have a direct feature or attribute related to Notebook Direct Internet Access. Therefore, it's not possible to check for Notebook Direct Internet Access in DynamoDB using AWS CLI. However, you can check for other security misconfigurations such as unencrypted tables or tables with public access. 
+
+   Please note that Notebook Direct Internet Access is a feature related to Jupyter notebooks, which are often used in data science and machine learning workflows, and not directly related to DynamoDB. If you are using AWS SageMaker for Jupyter notebooks, you can check the direct internet access setting of your notebook instances using the following AWS CLI command:
+
+   ```
+   aws sagemaker describe-notebook-instance --notebook-instance-name <YourNotebookInstanceName>
+   ```
+   Replace `<YourNotebookInstanceName>` with the name of your SageMaker notebook instance. This command will return a JSON object that contains detailed information about the notebook instance, including the `DirectInternetAccess` attribute. If the value of this attribute is `Enabled`, it means that the notebook instance has direct internet access.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check for Notebook Direct Internet Access in DynamoDB using Python scripts, you can use the Boto3 library, which allows you to directly create, update, and delete AWS services from your Python scripts. Here are the steps:
+
+1. **Import the Boto3 library in Python:**
+   Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then import it in your script:
+   ```python
+   import boto3
+   ```
+
+2. **Create a DynamoDB resource and client:**
+   You can create a DynamoDB resource and client using your AWS credentials. Replace 'your_access_key', 'your_secret_key', and 'your_region' with your actual AWS credentials.
+   ```python
+   dynamodb = boto3.resource('dynamodb', aws_access_key_id='your_access_key', aws_secret_access_key='your_secret_key', region_name='your_region')
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>